src/main/java/com/patina/codebloom/common/db/models/achievements/Achievement.java

```
@@ -0,0 +1,50 @@
+package com.patina.codebloom.common.db.models.achievements;
+
+import java.time.OffsetDateTime;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotNull;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+
+@Builder
+@Getter
+@Setter
+@ToString
+@EqualsAndHashCode
+@NoArgsConstructor
+@AllArgsConstructor
+public class Achievement {
+    @Schema(requiredMode = Schema.RequiredMode.REQUIRED)
+    private String id;
+
+    @NotNull
```

TLDR: Don't use `@NotNull` in db model classes: it may cause some unintended effects.

Long version: `@NotNull` is going to do a validation on the server level, which could cause some unintended behaviors when we don't expect it. Historically, we set allow vs disallow `NULL` in the database, and simply pass in any value (`null` or not) from the backend into the server and let the database throw an exception if it's valid or not.